include/small/Conv2DLayer.hpp

```diff
@@ -40,13 +40,13 @@ namespace detail
         BufferT          &m_packed_bias)
     {
         // ============ Filter weights ===========
-        if (filters.size() <
+        if (filters.size() !=   /// @todo consider allowing larger filter buffers??
```

Comment:
Is there a case where we might need a larger buffer size? Generally I feel like letting the filter buffers be larger might cause issues when indexing passed the _effective_ buffer size. I would prefer to get a seg fault if that indexing error would occur.